lib/reporting/reports/enterprise_fee_summary/enterprise_fees_with_tax_report_by_producer.rb

```diff
@@ -119,22 +119,47 @@ def join_tax_rate
           end
         end
 
+        # For every supplier, we want to show the enteprise fees
+        # applied to at least one of his products
         def join_supplier
           proc do |item|
             order = item[:order]
-            order
-              .line_items
-              .map(&:supplier_id)
-              .filter(&method(:supplier_ids_filter))
-              .map do |supplier_id|
+
+            filtered_line_items(order)
+              .map do |line_item|
                 {
                   tax_rate_id: item[:tax_rate_id],
-                  enterprise_fee_id: item[:enterprise_fee_id],
-                  supplier_id: supplier_id,
+                  enterprise_fee_id: if item[:enterprise_fee_id].in?(
+                    enterprise_fees_per_variant(order)[line_item.variant]
+                  )
+                                       item[:enterprise_fee_id]
+                                     end,
+                  supplier_id: line_item.supplier.id,
                   order: order
                 }
```

Comment:
instead, I suggest filtering and then mapping each item.

Reply:
Ok, I think my suggestion of merging is cleaner (no need to re-declare the `tax_rate_id` and `enterprise_fee_id` that are copied across). Actually, it might have been even cleaner and more efficient by editing the line_item hash directly (eg `line_item[:supplier_id] = line_item.supplier.id`)

But this works. Hopefully it's not too hard to optimise in the future if we need to.

Reply:
Sorry, I misunderstood your earlier comment. 

I was mapping before filtering, but I fixed that now.
 
Also, you're right, it'll be better to avoid using mapping and to modify the line_item objects directly.
I tried to do that, but it seems not possible. Please let me know what you think. 
```ruby
        def join_supplier
          proc do |item|
            order = item[:order]
            enterprise_fees_per_variant = enterprise_fees_per_variant(order)
            filtered_line_items(order)
              .filter do |line_item|
                item[:enterprise_fee_id].in?(
                  enterprise_fees_per_variant[line_item.variant]
                )
              end
              .each do |line_item|
                line_item[:tax_rate_id]= item[:tax_rate_id]
                line_item[:enterprise_fee_id]= item[:enterprise_fee_id]
                line_item[:supplier_id]= line_item.supplier.id
                line_item[:order]= order
              end
          end
        end
``` 
![image](https://github.com/openfoodfoundation/openfoodnetwork/assets/16113997/df166c56-eaff-4066-81fd-2675e5c8f4e3)

Reply:
Ah, thanks for trying. And sorry, I forgot that in this case, we're iterating over LineItems, not hashes. 
And other methods are relying on this one to aggregate some data into a hash, so we can't easily change this.

So lets move one :)